genesyscloud/architect_ivr/resource_genesyscloud_architect_ivr.go

```diff
@@ -0,0 +1,240 @@
+package architect_ivr
+
+import (
+	"context"
+	"fmt"
+	"log"
+	gcloud "terraform-provider-genesyscloud/genesyscloud"
+	"terraform-provider-genesyscloud/genesyscloud/consistency_checker"
+	"terraform-provider-genesyscloud/genesyscloud/util/resourcedata"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+
+	resourceExporter "terraform-provider-genesyscloud/genesyscloud/resource_exporter"
+	"terraform-provider-genesyscloud/genesyscloud/util/lists"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v112/platformclientv2"
+)
+
+const maxDnisPerRequest = 50
+
+// getAllIvrConfigs retrieves all architect IVRs and is used for the exporter
+func getAllIvrConfigs(ctx context.Context, clientConfig *platformclientv2.Configuration) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
+	resources := make(resourceExporter.ResourceIDMetaMap)
+	ap := getArchitectIvrProxy(clientConfig)
+
+	allIvrs, err := ap.getAllArchitectIvrs(ctx, "")
+	if err != nil {
+		return nil, diag.Errorf("failed to get architect ivrs: %v", err)
+	}
+
+	for _, entity := range *allIvrs {
+		resources[*entity.Id] = &resourceExporter.ResourceMeta{Name: *entity.Name}
+	}
+	return resources, nil
+}
+
+// createIvrConfig is used by the resource to create a Genesys Cloud Architect IVR
+func createIvrConfig(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	name := d.Get("name").(string)
+	description := d.Get("description").(string)
+	openHoursFlowId := gcloud.BuildSdkDomainEntityRef(d, "open_hours_flow_id")
+	closedHoursFlowId := gcloud.BuildSdkDomainEntityRef(d, "closed_hours_flow_id")
+	holidayHoursFlowId := gcloud.BuildSdkDomainEntityRef(d, "holiday_hours_flow_id")
+	scheduleGroupId := gcloud.BuildSdkDomainEntityRef(d, "schedule_group_id")
+	divisionId := d.Get("division_id").(string)
+	dnis := lists.BuildSdkStringList(d, "dnis")
```

Comment:
A `buildIvrConfigFromResourceData` function might help duplicating a lot of this in the update function